Lib/pathlib.py

```diff
@@ -664,8 +560,9 @@ def relative_to(self, *other):
         else:
             to_abs_parts = to_parts
         n = len(to_abs_parts)
-        cf = self._flavour.casefold_parts
-        if (root or drv) if n == 0 else cf(abs_parts[:n]) != cf(to_abs_parts):
+        def nc(parts):
+            return [self._flavour.normcase(p) for p in parts]
+        if (root or drv) if n == 0 else nc(abs_parts[:n]) != nc(to_abs_parts):
```

Comment:
See commend above about `nc()` to decide whether to take this or do something else.

```suggestion
        if not n:
            is_error = root or drive
        else:
            nc_abs_parts = list(map(self._flavour.normcase, abs_parts[:n]))
            nc_to_abs_parts = list(map(self._flavour.normcase, to_abs_parts[:n]))
            is_error = ns_abs_parts != nc_to_abs_parts
        if is_error:
```